designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Container.kt

```diff
@@ -0,0 +1,229 @@
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Divider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.geometry.CornerRadius
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.PathEffect
+import androidx.compose.ui.graphics.drawscope.Stroke
+import org.hisp.dhis.mobileui.designsystem.theme.Outline
+import org.hisp.dhis.mobileui.designsystem.theme.Radius
+import org.hisp.dhis.mobileui.designsystem.theme.Spacing
+import org.hisp.dhis.mobileui.designsystem.theme.SurfaceColor
+import org.hisp.dhis.mobileui.designsystem.theme.TextColor
+
+/**
+ * DHIS2 ColumnComponentContainer wraps Material 3 [Column]
+ * has a default spacing between items of 10 dp
+ * vertical scroll enabled
+ * @param title is the value of the text to be shown for the row.
+ * @param content controls the content to be shown
+ */
+@Composable
+fun ColumnComponentContainer(
+    title: String? = null,
+    content: @Composable (() -> Unit)
+) {
+    Column(
+        verticalArrangement = Arrangement.spacedBy(Spacing.Spacing16),
+        modifier = Modifier.padding(Spacing.Spacing10).verticalScroll(rememberScrollState())
+    ) {
+        title?.let {
+            Text(title, style = MaterialTheme.typography.titleMedium)
+        }
+        Spacer(Modifier.size(Spacing.Spacing4))
+        content()
+    }
+}
+
+/**
+ * DHIS2 RowComponentContainer wraps Material 3 [Row]
+ * @param title is the value of the text to be shown for the row.
+ * @param content controls the content to be shown
+ */
+
+@Composable
+fun RowComponentContainer(
+    title: String? = null,
+    content: @Composable (() -> Unit)
+) {
+    title?.let {
+        Text(title, style = MaterialTheme.typography.titleMedium)
+    }
+    Row(
+        horizontalArrangement = Arrangement.spacedBy(Spacing.Spacing10),
+        modifier = Modifier.padding(Spacing.Spacing10)
+    ) {
+        content()
+    }
+}
+
+/**
+ * DHIS2 Input Shell
+ * @param title is the value of the text to be shown.
+ * @param state controls the  state of the component. Will be Unfocused by default
+ * @param primaryButton controls the primary button composable if null will show nothing
+ * @param secondaryButton controls  action button composable, if null will show nothing
+ * @param legend controls the optional legend composable
+ * @param inputField controls the input field composable .
+ * @param supportingText controls the supporting text composable
+ */
+@Composable
+fun InputShell(
```

Comment:
I think that I'll extract everything of Inputshell into a custom class called InputShell and split it from container. What do you think?

Reply:
I thought about that and I agree it should be it's own separate component I'll extract it